BitWatch/HomeViewController.swift

```diff
@@ -97,227 +137,26 @@ class HomeViewController: UIViewController, ADBannerViewDelegate {
-            workSetting.font = UIFont(name: "Montserrat-Bold", size: 48);
-            workSetting.textColor = workPurple;
-            workSetting.text = "99:99";
-        
-        settingsView.addSubview(workLabel);
-        settingsView.addSubview(workSetting);
-        
-        // Create the Rest Setting Text Field
-        var restLabel: UILabel = UILabel(frame: CGRectMake(20, (mainView.frame.size.height * 0.5) - 30, settingsWidth - 40, 50));
-        restLabel.font = UIFont(name: "Montserrat-Bold", size: 22);
-        restLabel.textColor = restRed;
-        restLabel.text = "Rest";
-        var restSetting: UITextField = UITextField(frame: CGRectMake(20, mainView.frame.size.height * 0.5, settingsWidth - 40, 50));
-        restSetting.font = UIFont(name: "Montserrat-Bold", size: 48);
-        restSetting.textColor = restRed;
-        restSetting.text = "99:99";
-        
-        settingsView.addSubview(restLabel);
-        settingsView.addSubview(restSetting);
-        
-        // Create the Interval Setting Text Field
-        var intervalLabel: UILabel = UILabel(frame: CGRectMake(20, (mainView.frame.size.height * 0.8) - 30, settingsWidth - 40, 50));
-        intervalLabel.font = UIFont(name: "Montserrat-Bold", size: 22);
-        intervalLabel.textColor = intervalGreen;
-        intervalLabel.text = "Intervals";
-        intervalSetting = UITextField(frame: CGRectMake(20, mainView.frame.size.height * 0.8, settingsWidth - 40, 50));
-        intervalSetting.font = UIFont(name: "Montserrat-Bold", size: 48);
-        intervalSetting.textColor = intervalGreen;
-        intervalSetting.text = "9999";
-        
-        settingsView.addSubview(intervalLabel);
-        settingsView.addSubview(intervalSetting);
-        
-        mainView.addSubview(settingsView);
-    }
-    
-    
-    // *********************************
-    // Add the Menu Button to the View
-    // *********************************
-    
-    func addMenuButtonToView(view: UIView){
-        var menuButton: UIButton = UIButton(frame: CGRectMake(width - 35, 10, 25, 25));
-        var menuImage: UIImage! = UIImage(named: "menu-icon");
-        menuButton.setBackgroundImage(menuImage, forState: UIControlState.Normal);
-        menuButton.addTarget(self, action: "menuClicked", forControlEvents: UIControlEvents.TouchUpInside);
-        view.addSubview(menuButton);
-    }
-    
-    func menuClicked() {
-        if( menuOpen ){
-            mainView.setContentOffset(CGPointMake(0, 0), animated: true);
-            menuOpen = false;
-        } else {
-            mainView.setContentOffset(CGPointMake(width - 45, 0), animated: true);
-            menuOpen = true;
-        }
-    }
-    
-    
-    // *********************************
-    // Handle the Initial Countdown
-    // *********************************
-    
     func beginCountdown() {
-        // Hide the Begin Button
```

Comment:
To me this is a big reason for the refactor. The function BeginCountDown() should do just that, begin a countdown. it's a member of the controller and shouldn't be concerned with messy view setups. Plus, if you look at this every time you start the timer you are creating a new label and adding it to the homeview it becomes error prone when you repeat timers.